libcipher/caesar.py

```diff
@@ -0,0 +1,112 @@
+__author__ = 'kskrishnasangeeth'
+
+import string
+
+
+class Error(Exception):
+    pass
+
+
+class WrongKey(Error):
+    """ class created for throwing Exception in the
+        event of wrong key being supplied.
+
+    Attributes:
+        msg (str): Explanation on why key is not allowed.
+        key (int): Key that is supplied.
+
+    Args:
+        msg (str): Explanation on why key is not allowed.
+        key (int): Key that is supplied.
+
+   """
+    def __init__(self, msg, key):
+        self.msg = msg
+        self.key = key
+
+
+class NoKeyGiven(Error):
+    """ class created for throwing Exception in the
+        event of no key being supplied.
+
+    Attributes:
+        msg (str): Message to provide key for encryption.
+
+    Args:
+        msg (str): Message to provide key for encryption.
+
+    """
+    def __init__(self, msg):
+        self.msg = msg
+
+
+def cipher_helper(message, key):
+    """Function for modularising and removing some of the repeated code.
+
+    Args:
+        message (str): Message that is supplied for encryption/decryption.
+        key (int): Key used for encryption/decryption.
+
+    Returns:
+        message (str): Message in lowercase.
+
+    """
+    if key is None:
+        raise NoKeyGiven("Key is needed for Encryption.")
+
+    if key <= 0 or key >= 27:
+        raise WrongKey("Key Value should be between 0 and 26. Got Key as", key)
+
+    return message.lower()
+
+
+def encrypt(message, key=None):
+    """Returns the encrypted message for the string passed as argument.
+
+    Uses a random key for generating the cipher text
+
+    Args:
+        message (str): The message to be encrypted.
+        key (int): The key based on which encryption is performed.
+
+    Returns:
+        str: the encrypted message.
+
+    """
+    cipher_text = ""
+    msg = cipher_helper(message, key)
+    for each_char in msg:
+        char_pos = string.ascii_lowercase.find(each_char)
+        if char_pos == -1:
+            cipher_text += each_char
+            continue
+        else:
+            char_pos = (char_pos + key) % 26
+            cipher_text += string.ascii_lowercase[char_pos]
+    return cipher_text
+
+
+def decrypt(message, key):
```

Comment:
> A function validate_key() that checks and throws an error would be nice

I would actually prefer to leave them alone in the individual functions actually. Any reason to move them into another `validate_key` function? The purpose of those checks IMO is better if left with their respective functions. Having a `validate_key` functions means that we'll have to send a key indicating what type of thing to check for. Eg: 

```python
def validate_key(fn_type, key):
  if fn_type.strip().upper() == "ENCRYPT":
    # do something
  elif fn_type.strip().upper() == "DECRYPT":
    # do something

def encrypt(...):
  validate_key("encrypt", ...)

def decrypt(...):
  validate_key("decrypt", ...)
```

Which doesn't look all that elegant to me. 

> And BTW, just a ValueError would suffice. Do you need a separate exception class for this?

+1. Was thinking of this as well